### PR TITLE
#10 - Ability to edit users

### PR DESCRIPTION
**Pull request by hachesilva**
Added the ability to edit user's data from an interface similar to the user creation one.
Moved the socialStratum APIout of the controller.

### CODE REVIEW

File: src/views/users/edit.html [hunk 1]
```diff
@@ -0,0 +1,434 @@
+<div class="container" ng-controller="EditUsersController as editUsers">
+    <h1>{{editUsers.moduleName}}
+        <small>{{editUsers.pageName}}</small>
+    </h1>
+    {{editUsers}}
```
Comment (yayobyte) [inline src/views/users/edit.html]: This must be removed

File: src/js/app/modules/users/controllers/editUsers.controller.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        vm.validateFormData = function (){
+            // TODO: Convert this to a reusable component
+            // instead of a controller method
+
+            var userForm = $scope.editUsersForm;
+
+            for (var key in userForm) {
+                var elem = userForm[key];
+
+                // Look for chaged fields
```
Comment (yayobyte) [inline src/js/app/modules/users/controllers/editUsers.controller.js]: This is just a small comment: You mean CHANGED?

File: src/js/app/modules/users/controllers/editUsers.controller.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                if (typeof(elem) == 'object' && elem.hasOwnProperty('$dirty') && elem.$dirty === true) {
+                    var modelValue = null;
+
+                    // On these fields save just the id and not the whole object
+                    switch (elem.$name) {
```
Comment (yayobyte) [inline src/js/app/modules/users/controllers/editUsers.controller.js:75]: Not sure if I understand these empty "cases"

File: src/js/app/modules/users/controllers/editUsers.controller.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                            modelValue = elem.$modelValue;
+                    }
+
+                    vm.postObject[elem.$name] = modelValue;
+                }
+            }
+
+            UsersFactory.update(vm.postObject, {id: $routeParams.id}, function (response){
+                vm.postStatus = response;
+                vm.postStatus.message = "User successfully created";
```
Comment (yayobyte) [inline src/js/app/modules/users/controllers/editUsers.controller.js]: This should be better to see a "User successfully updated" or something similar that mentions update not creation

File: src/js/app/modules/users/controllers/editUsers.controller.js (hunk 3 above)
Comment (hachesilva) [inline src/js/app/modules/users/controllers/editUsers.controller.js:75]: Empty cases in switch statements means "for all of these cases we need a single action".

**Comment by hachesilva** [comment]
Done addressing @yayobyte's comments

**Comment by yayobyte** [comment]
I took this branch. Pull request closed

**Pull request closed without merging**